### PR TITLE
Support verification report output formats in the Plugin Verifier

### PR DESCRIPTION
**Pull request by novotnyr**
# Support verification report output formats

In the Plugin Verifier task, allow to specify plugin verification report formats, such as Markdown or HHTML

## Description

Allow to specify verification report formats that are produced by the Plugin Verifier. By default, only HTML verification output format is specified. However, Plugin Verifier allows multiple formats which might be included explicitly.

```
runPluginVerifier {
    verificationReportsFormats = [ "markdown", "plain" ]
}
```

## Motivation and Context

Plugins will be able to emit Markdown verification reports with the help of the Gradle IntelliJ Plugin, and later can be integrated with additional analysis tools, such as GitHub pipelines.

## How Has This Been Tested

Integration tests are provided in the source code.

## Types of changes

- [ ] Docs change / refactoring / dependency upgrade
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist

- [x] I have read the [**CONTRIBUTING**](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/CONTRIBUTING.md) document.
- [x] My code follows the code style of this project.
- [x] My change requires a change to the [documentation](https://plugins.jetbrains.com/docs/intellij/tools-gradle-intellij-plugin.html).
- [ ] I have updated the documentation accordingly.
- [ ] I have included my change in the [**CHANGELOG**](https://github.com/JetBrains/gradle-intellij-plugin/blob/master/CHANGELOG.md).
- [x] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

File: src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt [hunk 1]
```diff
@@ -687,6 +687,7 @@ abstract class IntelliJPlugin : Plugin<Project> {
             verificationReportsDir.convention(
                 project.layout.buildDirectory.dir("reports/pluginVerifier").map { it.asFile.canonicalPath }
             )
+            verificationReportsFormats.convention(listOf("plain", "html"))
```
Comment (hsz) [inline src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt:690]: Those were supposed to be enum values

File: src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt [hunk 2]
```diff
@@ -135,6 +139,98 @@ class RunPluginVerifierTaskSpec : IntelliJPluginSpecBase() {
...
+        writePluginXmlFile()
+        buildFile.groovy(
+            """
+            version = "1.0.0"
+            
+            runPluginVerifier {
+                verificationReportsFormats = [ "markdown", "plain" ]
+                verificationReportsDir = "${'$'}{project.buildDir}/foo"
+                ideVersions = ["IC-2020.2.3"]
+                verifierPath = "/tmp/verifier-cli.jar"
```
Comment (hsz) [inline src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt:154]: Why `/tmp/`?

File: src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt [hunk 3]
```diff
@@ -135,6 +139,98 @@ class RunPluginVerifierTaskSpec : IntelliJPluginSpecBase() {
...
+        build(RUN_PLUGIN_VERIFIER_TASK_NAME).let { buildResult ->
+            val reportsDirectory = file("build/foo")
+            val directory = reportsDirectory.canonicalPath
+            assertContains("Verification reports directory: $directory", buildResult.output)
+            val ideDirs = reportsDirectory.listFiles()
+            if (ideDirs.isEmpty()) {
+                fail("Verification reports directory not found")
+            }
+            val ideVersionDir = ideDirs.first()
+            val markdownReportFiles = ideVersionDir.listFilesOrdered { it.extension == "markdown" }
```
Comment (hsz) [inline src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt:168]: `.markdown`??

**Review by hsz — CHANGES_REQUESTED** [review]

File: src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt (hunk 2 above)
Comment (novotnyr) [reply src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt:154]: Will be removed when the PV is properly released.

File: src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt [hunk 4]
```diff
@@ -3,12 +3,16 @@
...
+import org.gradle.kotlin.dsl.support.listFilesOrdered
 import org.jetbrains.intellij.IntelliJPluginConstants.BUILD_PLUGIN_TASK_NAME
 import org.jetbrains.intellij.IntelliJPluginConstants.PLUGIN_VERIFIER_REPOSITORY
 import org.jetbrains.intellij.IntelliJPluginConstants.RUN_PLUGIN_VERIFIER_TASK_NAME
 import org.jetbrains.intellij.IntelliJPluginSpecBase
+import org.junit.Assert.assertTrue
+import org.junit.Assert.fail
 import java.net.URL
 import kotlin.test.Test
+import kotlin.test.assertEquals
```
Comment (YannCebron) [inline src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt:15]: use all assert from junit?

Comment (novotnyr) [reply src/test/kotlin/org/jetbrains/intellij/tasks/RunPluginVerifierTaskSpec.kt]: Consolidated to JUnit.

File: src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt (hunk 1 above)
Comment (novotnyr) [reply src/main/kotlin/org/jetbrains/intellij/IntelliJPlugin.kt]: Declared constants in `org.jetbrains.intellij.IntelliJPluginConstants`.

**Review by hsz — APPROVED** [review]

**Pull request merged**